### ISSUE TITLE
Support 'Name' property in Custom Resource

### ISSUE THREAD

**Issue by aparker-cityworks** [awslabs/aws-instance-scheduler#71]
I have been able to create new Schedules using the Custom Resource in a separate stack.  The name of the Schedule is created by concatenating the Stack name with the name of the resource.  This makes the name rather long and not very good for using as a tag value.

I assumed that I could define the Name property on the custom resource to over ride that behavior. However I got this error when I tried that.
```
ERROR : Name is not a valid property for a schedule, valid schedule properties are Description, Enforced, Metrics, OverrideStatus, Overwrite, Periods, StopNewInstances, Timezone, UseMaintenaceWindow, ServiceToken, Timeout
```
Could you please add the ability to override the name of the schedule by specifying the Name property on the Custom Resource?  I see that you are getting ready to release a new version soon.  Please include this in the upcoming release!

**Comment by arieleeuw** [comment]
Hi,
In the upcoming release the custom resource has a property named "NoStackPrefix" that suppresses the stack name prefix. The schedule will take the name of the resource without the prefix if the value is set to True
Hope this works for you

**Comment by aparker-cityworks** [comment]
That is better than the previous behavior, but still a bit limited.  For example, it's not possible to use a dash (-) in the name of a CloudFormation resource.

I still think a better solution would be to support the Name parameter on the Custom Resource.

**Comment by aparker-cityworks** [comment]
From the CloudFormation documentation https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/resources-section-structure.html

> The logical ID must be alphanumeric (A-Za-z0-9) and unique within the template.

That greatly limits what you can name a Schedule when using the Custom Resource.  If one could specify a Name property, then there would be no limitation when using the Custom Resource.  Is it possible for this to be implemented before the pending release?

**Comment by arieleeuw** [comment]
The next update will have a Name property to explicitly set the name of the schedule and a NoStackPrefix to disable the name of the Stack as a prefix

**Comment by aparker-cityworks** [comment]
That will be perfect.  Do you have an estimation of release time frame?

**Comment by shsenior** [comment]
Fixed in v1.2.0
